core/unit_test/view/TestViewEqualityOperator.hpp

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright Contributors to the Kokkos project
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Macros.hpp>
+#ifdef KOKKOS_ENABLE_EXPERIMENTAL_CXX20_MODULES
+import kokkos.core;
+import kokkos.core_impl;
+#else
+#include <Kokkos_Core.hpp>
+#endif
+#include <cstddef>
+
+// Test View equality operators
+// View equivalence defined as matching value_type,,
+// layout_type, memory_space, rank, span, data pointers and extents
+
+namespace {
+
+// TODO: Replace SFINAE with concepts when we have have ArrayLayout and
+// MemoryTraits
```

Comment:
What are you referring to?  "ArrayLayout" and "MemoryTraits" concepts?
You could already use concepts here if you like (`requires (Kokkos::is_array_layout_v<T>)`)
In any case I don't think it is worth having a TODO here.